cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -1091,8 +1153,17 @@ struct GroupedMeanImpl : public GroupedReducingAggregator<Type, GroupedMeanImpl<
   }
 };
 
+struct GroupedMeanNullImpl final : public GroupedNullImpl {
+  std::shared_ptr<DataType> out_type() const override { return float64(); }
+
+  void output_empty(const std::shared_ptr<Buffer>& data) override {
+    std::memset(data->mutable_data(), 0, num_groups_ * sizeof(double));
```

Comment:
I'd rather use `std::fill_n(double*, ...)` here for clarity.

Reply:
@pitrou, `memset` is replaced with `fill_n`.